snapcraft/commands/clean.py

```diff
@@ -62,7 +57,8 @@ def main(argv=None):
             not os.listdir(common.get_partsdir())):
         os.rmdir(common.get_partsdir())
 
-    clean_stage = not args['PART'] or part_names == set(args['PART'])
+    parts_match = set(config.part_names) == set(args['PART'])
+    clean_stage = not args['PART'] or parts_match
     if clean_stage and os.path.exists(common.get_stagedir()):
```

Comment:
This deserves a comment. "Delete the stage dir only if all the parts were cleaned"

did I get it right?